force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,10 +175,38 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
+        // Query the Account Id based on the provided accountName
+        List<Account> accList = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        Account acc;
+        // If Account doesn't exist, create a new one
+        if (acc == null) {
+            acc = new Account(Name = accountName);
+            insert acc;
+        }
```

Comment:
The way you've written this code, `acc` will always be null. You should check the results from your query first, then decide if you want to assign acc or leave it not initialized.